build-android/app/src/main/java/com/google/firebase/codelab/friendlychat/MyButtonObserver.kt

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.firebase.codelab.friendlychat
+
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Button
+
+class MyButtonObserver(private val button: Button) : TextWatcher {
+    override fun onTextChanged(charSequence: CharSequence, start: Int, count: Int, after: Int) {
+        button.isEnabled = charSequence.toString().trim { it < ' ' }.isNotEmpty()
```

Comment:
I had no idea that `trim` took a callback/predicate in Kotlin! Just curious how is this different from `trim(' ')` ... does the `<` remove other kinds of whitespace too?

Reply:
Pfft I guess so, but doesn't seem important. I copy-pasted that over from the written Java, so not sure what the original intention was here. I'm just going to use "trim()" for simplicity